### PR TITLE
rename and deprecate RPCs, rename vars and files

### PR DESCRIPTION
**Pull request by deepthi**
Signed-off-by: deepthi <deepthi@planetscale.com>

### CODE REVIEW

File: proto/replicationdata.proto [hunk 1]
```diff
@@ -21,20 +21,23 @@ option go_package = "vitess.io/vitess/go/vt/proto/replicationdata";
...
   uint32 master_server_id = 11;
-  bool slave_io_running = 2;
-  bool slave_sql_running = 3;
-  uint32 seconds_behind_master = 4;
-  string master_host = 5;
-  int32 master_port = 6;
-  int32 master_connect_retry = 7;
   string master_uuid = 12;
+  bool io_thread_running = 13;
+  bool sql_thread_running = 14;
```
Comment (sougou) [inline proto/replicationdata.proto:42]: Is this because it's a new feature? If so, how come the ids are this low? Were they unused before?

Comment (deepthi) [reply proto/replicationdata.proto:42]: This is renaming slave_io_running and slave_sql_running. The old names have been removed and marked as reserved, and replaced with new names.

Comment (deepthi) [reply proto/replicationdata.proto:42]: the extra diff is because the fields were not ordered by number and it was confusing.

Comment (sougou) [reply proto/replicationdata.proto:42]: I was asking about why you didn't continue to use the original numbers 2 & 3, like you did for the other cases.

Comment (deepthi) [reply proto/replicationdata.proto:42]: can I do that? I didn't think you could rename proto fields. I looked at the rest of the diff and the only other proto change is force_start_slave -> force_start_replication where also I reserved 3 and added 5.

Comment (deepthi) [reply proto/replicationdata.proto:42]: from offline discussion, it seems like the correct approach is to rename the fields. reserving and creating new fields won't work across versions.

**Comment by deepthi** [comment]
@sougou this is ready for re-review.
Here's how I tested the proto changes:
- Run vttablets in local example with new proto definitions
- Start vtctld with older build (master w/o this PR changes)
- test that `ShardReplicationStatuses` works correctly. This tests the changes to replicationdata protobuf. I added more output  locally to `ShardReplicationStatuses` to validate the values of IoThreadRunning and SqlThreadRunning. 
- test `ReparentTablet` and `PlannedReparentShard`. These test the changes to `SetMasterRequest` where `force_start_slave` is being renamed to `force_start_replication`. One of them sends the value as false and one as true, so both possibilities are tested.
- All these manual tests worked as expected. Verified by inspecting logs and cluster state.

**Pull request merged**